src/http_handlers/metrics.hpp

```diff
@@ -0,0 +1,127 @@
+// Copyright 2023 Memgraph Ltd.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt; by using this file, you agree to be bound by the terms of the Business Source
+// License, and you may not use this file except in compliance with the Business Source License.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+#pragma once
+
+#include <spdlog/spdlog.h>
+#include <boost/beast/http.hpp>
+#include <boost/beast/version.hpp>
+#include <json/json.hpp>
+
+#include "query/interpreter.hpp"
+#include "storage/v2/storage.hpp"
+
+namespace memgraph::http {
+
+namespace beast = boost::beast;
+namespace http = beast::http;
+
+struct MetricsResponse {
+ public:
+  nlohmann::json AsJson() {
+    auto metrics_response = nlohmann::json();
+    metrics_response["vertex_count"] = vertex_count;
+    metrics_response["edge_count"] = edge_count;
+    metrics_response["average_degree"] = average_degree;
+    metrics_response["memory_usage"] = memory_usage;
+    metrics_response["disk_usage"] = disk_usage;
+    return metrics_response;
+  }
+
+  uint64_t vertex_count;
+  uint64_t edge_count;
+  double average_degree;
+  uint64_t memory_usage;
+  uint64_t disk_usage;
+};
+
+template <typename TSessionData>
+class MetricsService {
+ public:
+  explicit MetricsService(TSessionData *data)
+      : db_(data->db), interpreter_context_(data->interpreter_context), interpreter_(data->interpreter_context) {}
+  MetricsResponse GetMetrics() {
+    auto info = db_->GetInfo();
+    return MetricsResponse{.vertex_count = info.vertex_count,
+                           .edge_count = info.edge_count,
+                           .average_degree = info.average_degree,
+                           .memory_usage = info.memory_usage,
+                           .disk_usage = info.disk_usage};
+  }
+
+ private:
+  const storage::Storage *db_;
+  query::InterpreterContext *interpreter_context_;
```

Comment:
Where are those variables used? Also, even if you need both I think having interpreter is enough because you can fetch context from it.

Reply:
You get this from SessionData and I think this is different struct and that is how such struct is constructed

Reply:
Sorry can you explain, not sure what you meant with such struct is constructred

Reply:
db_ is used, but the data is the same one that is passed to session. I don't use the interpreter context though and the interpreter, I could remove it.